src/main/java/com/agnieszkapawska/flashcards/domain/facades/LearningFacade.java

```diff
@@ -0,0 +1,122 @@
+                          FlashcardsToRepeatService flashcardsToRepeatService,
+                          FlashcardsToRefreshService flashcardsToRefreshService,
+                          ModelMapper modelMapper) {
+        this.flashcardsToLearnService = flashcardsToLearnService;
+        this.flashcardService = flashcardService;
+        this.flashcardsToRepeatService = flashcardsToRepeatService;
+        this.flashcardsToRefreshService = flashcardsToRefreshService;
+        this.modelMapper = modelMapper;
+    }
+
+    public List<FlashcardGetResponseDto> getFlashcardsToLearnByUserId(Long userId) {
+        FlashcardsToLearn flashcardsToLearn = flashcardsToLearnService.findByUserId(userId).get();
+        ArrayList<Flashcard> flashcards = new ArrayList<>(flashcardsToLearn.getFlashcards());
+        return flashcards.stream()
+                .map(flashcard -> modelMapper.map(flashcard, FlashcardGetResponseDto.class))
+                .collect(Collectors.toList());
+    }
+
+    public void markAnswer(Answer answer) {
+        Long flashcardId = Long.parseLong(answer.getFlashcardId());
+        shouldReturnFromMethod = false;
+        Flashcard flashcard = flashcardService.findById(flashcardId);
+
+        flashcardsToLearnService.findByFlashcardId(flashcardId)
+                .ifPresent(flashcardsToLearn -> {
+                    markAnswer(answer, flashcard, flashcardsToLearn, flashcardsToLearnService, flashcardsToRepeatService);
+                    flashcardService.saveFlashcard(flashcard);
+                });
+        if (shouldReturnFromMethod) {
+            return;
+        }
+
+        flashcardsToRepeatService.findByFlashcardIs(flashcardId)
+                .ifPresent(flashcardsToRepeat -> {
+                    markAnswer(answer, flashcard, flashcardsToRepeat, flashcardsToRepeatService, flashcardsToRefreshService);
+                    flashcardService.saveFlashcard(flashcard);
+                });
+        if (shouldReturnFromMethod) {
+            return;
+        }
+
+        flashcardsToRefreshService.findByFlashcardId(flashcardId)
+                .ifPresent(flashcardsToRefresh -> {
+                    markAnswer(answer, flashcard, flashcardsToRefresh, flashcardsToRefreshService, flashcardsToRefreshService);
+                    flashcardService.saveFlashcard(flashcard);
+                });
+    }
+
+    private void markAnswer(
+            Answer answer,
+            Flashcard flashcard,
+            FlashcardsStorage flashcardsStorage,
+            FlashcardsStorageService currentFlashcardsStorageService,
+            FlashcardsStorageService futureFlashcardsStorageService) {
+
+        if(answer.getIsCorrect()) {
+            if (flashcard.getCorrectAnswerCounter() < 2) {
+                flashcard.setCorrectAnswerCounter(flashcard.getCorrectAnswerCounter() + 1);
+            } else {
+                if(flashcardsStorage instanceof FlashcardsToRefresh) {
+                    shouldReturnFromMethod = true;
```

Comment:
Maybe instead of using instance field we could just make markAnswer returning Bool? :)